codegen/src/main/java/software/amazon/awssdk/codegen/docs/Factory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.codegen.docs;
+
+import java.util.function.BiFunction;
+import software.amazon.awssdk.codegen.model.intermediate.IntermediateModel;
+import software.amazon.awssdk.codegen.model.intermediate.OperationModel;
+
+/**
+ * Factory interface to reduce the verbosity of using BiFunction with type args everywhere.
+ */
+@FunctionalInterface
+interface Factory extends BiFunction<IntermediateModel, OperationModel, OperationDocProvider> {
```

Comment:
Sure thing, It was originally an inner interface but I must have factored it out at some point.